### ISSUE TITLE
BUG: dataframe.fillna doesn't work

### ISSUE THREAD

**Issue by Yan-Daojiang** [pandas-dev/pandas#41149]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

<img width="825" alt="截屏2021-04-25 下午2 46 04" src="https://user-images.githubusercontent.com/34571894/115983689-476ee880-a5d5-11eb-8f01-9ccc7d04659f.png">

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : 2cb96529396d93b46abab7bbc73a208e708c642e
python           : 3.8.8.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.3.0
Version          : Darwin Kernel Version 20.3.0: Thu Jan 21 00:06:51 PST 2021; root:xnu-7195.81.3~1/RELEASE_ARM64_T8101
machine          : arm64
processor        : arm
byteorder        : little
LC_ALL           : None
LANG             : zh_CN.UTF-8
LOCALE           : zh_CN.UTF-8

pandas           : 1.2.4
numpy            : 1.18.5
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.0.1
setuptools       : 56.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.22.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.4.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.6.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None
</details>

**Comment by phofl** [comment]
Hi,

```python
df = DataFrame([[np.nan, np.nan], [3, 4], [np.nan, 5]], columns=["a", "b"])
df.fillna(0)
```
 
works for me. If you think this is specific to your DataFrame please provide a copy and pastable example

**Comment by Yan-Daojiang** [comment]
Hi, 
I just used example you give to me, like this:
```python
import pandas
import numpy as np
df = pandas.DataFrame([[np.nan, np.nan], [3, 4], [np.nan, 5]], columns=["a", "b"])
df.fillna(0)
```
and the output is:
     a    b
0  NaN  NaN
1  3.0  4.0
2  NaN  5.0

<img width="924" alt="截屏2021-04-25 下午11 29 01" src="https://user-images.githubusercontent.com/34571894/115999433-46ae7480-a61e-11eb-8055-7a4aabcf81e6.png">

the output of `pd.show_versions()` I have provided for you, and if you need more information, please tell me!

Thank you!

**Comment by phofl** [comment]
I have the same numpy and pandas versions, but ubuntu os. Maybe someone can try on mac?

**Comment by Yan-Daojiang** [comment]
Oh, thanks, it really works on my Win10 os, but can't work on my Macbook with apple silicon.
Maybe I will find some way to replace `fillna()` on my macOS.

**Comment by phofl** [comment]
Maybe an arm problem. Could you upgrade to the newest numpy to check if this persists? Just curious

**Comment by savoskin0502** [comment]
Hi, 
Works on my macOS. 
![image](https://user-images.githubusercontent.com/35302214/116006602-c42db100-a62d-11eb-8ddb-9898f7cd13ec.png)
INSTALLED VERSIONS
------------------
commit           : f2c8480af2f25efdbd803218b9d87980f416563e
python           : 3.8.8.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.4.0
Version          : Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8

pandas           : 1.2.3
numpy            : 1.20.1
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 20.2.3
setuptools       : 49.2.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.21.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 2.0.1
xlwt             : None
numba            : None

**Comment by phofl** [comment]
I think the arm processor is the problem, not the mac itself

**Comment by Yan-Daojiang** [comment]
I just update numpy from 1.18.5 to 1.20.0,  the problem truly disappeared. 
<img width="1006" alt="截屏2021-04-26 上午7 37 40" src="https://user-images.githubusercontent.com/34571894/116013336-4e900800-a662-11eb-98c2-e7d086244c54.png">

Now I am not sure wheater the arm processor is the problem or the numpy.

**Comment by fangchenli** [comment]
How did you install `numpy=1.18.5` on M1 Mac? `numpy` only has wheels for M1 after `1.20.0`. I remember I had to build `numpy` from master when I first got the M1 Mac last year.

**Comment by Yan-Daojiang** [comment]
Because I need to use TensorFlow, so I install `numpy==1.18.5` from [tensorflow_macos](https://github.com/apple/tensorflow_macos)

**Comment by lithomas1** [comment]
@Yan-Daojiang Is it OK to close this issue since it appears you resolved it? Thanks.

**Comment by lithomas1** [comment]
Fillna depends on isna, which depends on numpy. Given that numpy 1.20.0 fixed the issue in #42443, I think this can be closed. Please feel free to open another issue if you still have problems.
